### ISSUE TITLE
Cant authenticate to Google Drive

### ISSUE THREAD

**Issue by rudgej** [sabeechen/hassio-google-drive-backup#833]
New user, added the app and then opened the web UI.

Clicked Authenticate with Google Drive.

Logged in. Clicked the "Send Credentials" button and get unauthorised.

Using the manual method I get 

"Couldn't Save the credentials - did you copy the whole string"

and

Got Unexpected HTTP400 euro : Bad Request.

I have used 2 gmail accounts and both safari and chrome. 

All results are the same.

For whatever reason the authentication seems to be broken.

**Comment by buddy9880** [comment]
I'm having the same issue.  Here is my log file.
[home-assistant-google-drive-backup.log](https://github.com/sabeechen/hassio-google-drive-backup/files/11305564/home-assistant-google-drive-backup.log)

**Comment by tayanov** [comment]
+1

**Comment by sabeechen** [comment]
I think I've root caused this, and it seems to be caused by [this change](https://github.com/home-assistant/core/commit/b033232b06e14759bd886204e2efd3347a8384be) in Home Assistant.  URL's that go through Home Assistant no longer allow certain whitespace characters (like newlines).

If this is whats happening to you, you should see a message like this in Home Assistant's logs when you attempt to authorize the addon:
```log
[homeassistant.components.http.security_filter] Filtered a request with unsafe byte query string: /api/hassio_ingress/.../token?creds=<long_string_goes_here>&host=http%3A%2F%2F192.169.1.1%3A8123%2Ff0e7e328_hassio_google_drive_backup
```

Sorry it took me a while to root cause this, the problem is obvious now but its difficult to track down all the places HA might log errors.

As a workaround until I release a fix, when you paste the authorization string into the addon make sure you remove any trailing or leading whitespace (newlines, tabs, etc) from what gets put into the box.  The whitespace looks to be what causes the problem.

**Comment by buddy9880** [comment]
This worked for me.  Thank you.

**Comment by sabeechen** [comment]
I just released v 0.110.4 with a fix for this issue.  If anyone continues to see problems authenticating after updating to 0.110.4, please let me know and reopen this issue so I can help.
